src/test/convert-to-single-host.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+import * as assert from 'yeoman-assert';
+import * as fs from "fs";
+import * as helpers from 'yeoman-test';
+import * as path from 'path';
+import { promisify } from "util";
+const readFileAsync = promisify(fs.readFile);
+const hosts = ["excel", "onenote", "outlook", "powerpoint", "project", "word"];
+const unexpectedManifestFiles = [
+    'manifest.excel.xml',
+    'manifest.onenote.xml',
+    'manifest.outlook.xml',
+    'manifest.powerpoint.xml',
+    'manifest.project.xml',
+    'manifest.word.xml',
+]
+
+// Test to verify converting a project to a single host
+// for Office-Addin-Taskpane Typescript project using Excel host
+describe('Office-Add-Taskpane-Ts projects', () => {
+    const expectedFiles = [
+        'package.json',
+        'manifest.xml',
+        'src/taskpane/taskpane.ts',
+    ]
+    const unexpectedFiles = [
+        'src/taskpane/excel.ts',
+        'src/taskpane/onenote.ts',
+        'src/taskpane/outlook.ts',
+        'src/taskpane/powerpoint.ts',
+        'src/taskpane/project.ts',
+        'src/taskpane/word.ts'
+    ]
+    let answers = {
+        projectType: "taskpane",
+        scriptType: "TypeScript",
+        name: "TaskpaneProject",
+        host: hosts[0]
+    };
+
+    describe('Office-Add-Taskpane project', () => {
+        before((done) => {
+            helpers.run(path.join(__dirname, '../app')).withPrompts(answers).on('end', done);
+        });
+
+        it('creates expected files', (done) => {
+            assert.file(expectedFiles);
+            assert.noFile(unexpectedFiles);
+            assert.noFile(unexpectedManifestFiles);
+            done();
+        });
+    });
+
+    describe('Package.json is updated appropriately', () => {
+        it('Package.json is updated properly', async () => {
+            const data: any = await readFileAsync("package.json", 'utf8');
+            let content = JSON.parse(data);
+            assert.equal(content.config["app-to-debug"], hosts[0]);
+
+            // Verify host-specific sideload and unload sripts have been removed
+            let unexexpectedScriptsFound: boolean = false;
+            Object.keys(content.scripts).forEach(function (key) {
+                if (key.includes("sideload:") || key.includes("unload:")) {
+                    unexexpectedScriptsFound = true;
+                }
+            });
+            assert.equal(unexexpectedScriptsFound, false);
+        });
+    });
+
+    describe('Manifest.xml is updated appropriately', () => {
+        it('Manifest.xml is updated appropriately', async () => {
+            const expectedHostType = `<Host Name="Workbook"/>`;
+            const unexpectedHostContentTypes = [
```

Comment:
changing this to unexpectedHostTypes here and throughout

Reply:
You could simplify this to just check if there only expected hosts found. You don't really need to list unexpected hosts.

You can leverage office-addin-manifest package to parse the manifest. I've already updated it so you can get the hosts easily.

Reply:
But don't we want to be sure that all the host types other than the specified hosts have been removed from the manifest?  This ensures that we are in fact cleaning up the manifest entirely with convert-to-single-host

Reply:
OK never mind - I see what you mean here. I will just use office-addin-manifest info <manifestPath> to get the host